aspnetcore/fundamentals/minimal-apis/min-api-filters.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Filters in Minimal API applications
+author: rick-anderson
+description: Use filters in Minimal API applications
+ms.author: riande
+ms.date: 6/22/2022
+monikerRange: '>= aspnetcore-7.0'
+uid: fundamentals/minimal-apis/min-api-filters
+---
+# Filters in Minimal API apps
+
+By [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+Minimal API filters allow developers to implement business logic that supports:
+
+* Running code before the route handler.
+* Inspecting and modifying parameters provided during a route handler invocation.
+* Intercepting the response behavior of a route handler.
+
+Filters can be helpful in the following scenarios:
+
+* Validating the request parameters and body that are sent to an endpoint.
+* Logging information about the request and response.
+* Validating that a request is targeting a supported API version
+
+Filters can be registered by providing a [Delegate](/dotnet/csharp/programming-guide/delegates/) that takes a [`routeHandlerInvocationContext`](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Abstractions/src/RouteHandlerInvocationContext.cs) and returns  a [`RouteHandlerFilterDelegate`](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Abstractions/src/RouteHandlerFilterDelegate.cs). The `RouteHandlerInvocationContext` provides access to the `HttpContext` associated with the request and a `Arguments` list indicating the arguments passed to the handler in the order in which they appear in the argument list of the handler.
+
+[!code-csharp[](~/fundamentals/minimal-apis/min-api-filters/7samples/Filters/Program.cs?name=snippet1)]
+
+The preceding code:
+
+* Calls the `AddFilter` extension method to add a filter to the `/colorSelector/{color}` endpoint.
+* Returns the color specified except for the `Red`.
+* Returns [Results.Problem](xref:Microsoft.AspNetCore.Http.Results.Problem%2A) when the `/colorSelector/Red` is requested.
+* Uses `next` as the `RouteHandlerFilterDelegate` and `rhiContext` as the `routeHandlerInvocationContext`.
+
+The filter is run before the endpoint handler. When multiple `AddFilter` invocations are made on a handler:
+
+* The filters are executed in order of First In, First Out (FIFO) order.
+* The first filter registered runs first, the last filter registered runs last.
```

Comment:
In #25483 it states:
 When multiple AddFilter invocations are made on a handler, the filters will be executed in order of FILO order, so the first filter registered will run last.

In a previous preview when I tested it seems like they were FILO, but now they're FIFO

Reply:
Yeah, from the log it is also FIFO,

```
info: Filters.RouteFilters.ABCrouteFilters[0]
      Executing filter method Filters.RouteFilters.ArouteFilter
info: Filters.RouteFilters.ABCrouteFilters[0]
      Executing filter method Filters.RouteFilters.BrouteFilter
info: Filters.RouteFilters.ABCrouteFilters[0]
      Executing filter method Filters.RouteFilters.CrouteFilter
```